pkg/kds/context/context_test.go

```diff
@@ -0,0 +1,236 @@
+	stdcontext "context"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	system_proto "github.com/kumahq/kuma/api/system/v1alpha1"
+	config_manager "github.com/kumahq/kuma/pkg/core/config/manager"
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	core_system "github.com/kumahq/kuma/pkg/core/resources/apis/system"
+	"github.com/kumahq/kuma/pkg/core/resources/manager"
+	"github.com/kumahq/kuma/pkg/core/resources/model"
+	"github.com/kumahq/kuma/pkg/core/resources/registry"
+	core_store "github.com/kumahq/kuma/pkg/core/resources/store"
+	"github.com/kumahq/kuma/pkg/kds/context"
+	"github.com/kumahq/kuma/pkg/kds/reconcile"
+	"github.com/kumahq/kuma/pkg/plugins/resources/memory"
+	test_model "github.com/kumahq/kuma/pkg/test/resources/model"
+	zone_tokens "github.com/kumahq/kuma/pkg/tokens/builtin/zone"
+	"github.com/kumahq/kuma/pkg/tokens/builtin/zoneingress"
+	util_proto "github.com/kumahq/kuma/pkg/util/proto"
+)
+
+var _ = Describe("Context", func() {
+	Describe("GlobalProvidedFilter", func() {
+		var rm manager.ResourceManager
+		var predicate reconcile.ResourceFilter
+
+		clusterID := "cluster-id"
+		configs := map[string]bool{
+			config_manager.ClusterIdConfigKey: true,
+		}
+
+		BeforeEach(func() {
+			rm = manager.NewResourceManager(memory.NewStore())
+			predicate = context.GlobalProvidedFilter(rm, configs)
+		})
+
+		It("Should filter out dataplanes", func() {
+			// when
+			dp := &core_mesh.DataplaneResource{
+				Meta: &test_model.ResourceMeta{
+					Name: "dp-1",
+				},
+			}
+
+			// then
+			Expect(predicate(clusterID, dp)).To(BeFalse())
+		})
+
+		It("Should filter out configs if not in provided argument", func() {
+			// when
+			config1 := &core_system.ConfigResource{
+				Meta: &test_model.ResourceMeta{
+					Name: config_manager.ClusterIdConfigKey,
+				},
+			}
+			config2 := &core_system.ConfigResource{
+				Meta: &test_model.ResourceMeta{
+					Name: "config-which-should-be-filtered-out",
+				},
+			}
+
+			// then
+			Expect(predicate(clusterID, config1)).To(BeTrue())
+			Expect(predicate(clusterID, config2)).To(BeFalse())
+		})
+
+		It("Should filter out global secrets if they are not signing keys", func() {
+			// when
+			// zone ingress signing key
+			secret1 := &core_system.GlobalSecretResource{
+				Meta: &test_model.ResourceMeta{
+					Name: zoneingress.ZoneIngressSigningKeyPrefix + "-1",
+				},
+			}
+			// zone token signing key
+			secret2 := &core_system.GlobalSecretResource{
+				Meta: &test_model.ResourceMeta{
+					Name: zone_tokens.SigningKeyPrefix + "-1",
+				},
+			}
+			// zone token signing key
+			secret3 := &core_system.GlobalSecretResource{
+				Meta: &test_model.ResourceMeta{
+					Name: "some-other-global-secret",
+				},
+			}
+
+			// then
+			Expect(predicate(clusterID, secret1)).To(BeTrue())
+			Expect(predicate(clusterID, secret2)).To(BeTrue())
+			Expect(predicate(clusterID, secret3)).To(BeFalse())
+		})
```

Comment:
nit: I'd rewrite tests above for DescribeTable with Entries also everything from `It("Should filter resources with KDS Flags equal model.ProvidedByGlobal` could be another entries

Reply:
addressed